### PR TITLE
First upload with POM and Page Element

### PR DESCRIPTION
**Pull request by syringomyela**
1. Added Playwright
2. Created POM and Page Element files

### CODE REVIEW

**Review by Den4ik699 — CHANGES_REQUESTED** [review]

File: tests/demoQA/pages/inputs.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+function randomizeInput() {
+    const input = Math.random().toString(36).substring(2, 10)
+    return input;
```
Comment (Den4ik699) [inline tests/demoQA/pages/inputs.js:4]: we can use just `return Math.random().toString(36).substring(2, 10)`

File: tests/demoQA/pages/inputs.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+function randomizeEmail() {
+    const email = Math.random().toString(36).substring(2, 10) + '@mail.com'
+    return email;
```
Comment (Den4ik699) [inline tests/demoQA/pages/inputs.js:9]: we can use just return `Math.random().toString(36).substring(2, 10) + '@mail.com'`

File: tests/demoQA/pages/inputs.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+function randomizeInput() {
```
Comment (Den4ik699) [inline tests/demoQA/pages/inputs.js:2]: Let's give more understandable names to functions, for example `generateText()`

File: tests/testCases/negativeTC.spec.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const { test, expect } = require('@playwright/test');
+import { mainPage } from '../demoQA/pages/pageObject';
+import { filler } from '../demoQA/pages/inputs';
+
+test('Negative scenario', async({page}) => {
```
Comment (Den4ik699) [inline tests/testCases/negativeTC.spec.js:5]: Let's give more understandable names to tests, for example `Negative scenario: Enter incorrect email`

File: tests/testCases/negativeTC.spec.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+   
+    const pageQA = new mainPage(page);
+
+    await pageQA.goto();
+    await pageQA.fillTextBoxes(
+        filler.name, filler.fakeEmail, filler.currentAddress, filler.permanentAddress
+    );
+    await pageQA.pressSubmit();
+    
+    await expect(page.locator('#userEmail')).toHaveClass(/field-error/); //`Name:${filler.name}`
```
Comment (Den4ik699) [inline tests/testCases/negativeTC.spec.js:15]: We also should check the text of error

File: tests/testCases/positiveTC.spec.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+const { test, expect } = require('@playwright/test');
+import { mainPage } from '../demoQA/pages/pageObject';
+import { filler } from '../demoQA/pages/inputs';
+
+test ('Positive scenario', async ({page}) => {
+    
+    const pageQA = new mainPage(page);
+
+
```
Comment (Den4ik699) [inline tests/testCases/positiveTC.spec.js:9]: delete this spaces

File: tests/testCases/positiveTC.spec.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+const { test, expect } = require('@playwright/test');
+import { mainPage } from '../demoQA/pages/pageObject';
+import { filler } from '../demoQA/pages/inputs';
+
+test ('Positive scenario', async ({page}) => {
```
Comment (Den4ik699) [inline tests/testCases/positiveTC.spec.js:5]: Let's give more understandable names to tests, for example `Positive scenario: Fill form with correct data`

File: tests/testCases/positiveTC.spec.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+
+
+    await pageQA.goto();
+    await pageQA.fillTextBoxes(
+        filler.name, filler.email, filler.currentAddress, filler.permanentAddress
+    );
+    await pageQA.pressSubmit();
+
+    const outputName = await page.locator('#output #name').innerText();
```
Comment (Den4ik699) [inline tests/testCases/positiveTC.spec.js:17]: wrap this locators in Page Object

File: tests/testCases/positiveTC.spec.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    const outputEmail = await page.locator('#output #email').innerText();
```
Comment (Den4ik699) [inline tests/testCases/positiveTC.spec.js:18]: wrap this locators in Page Object

File: tests/testCases/positiveTC.spec.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    const outputCurrentAddress = await page.locator('#output #currentAddress').innerText();
```
Comment (Den4ik699) [inline tests/testCases/positiveTC.spec.js:19]: wrap this locators in Page Object

File: tests/testCases/positiveTC.spec.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+    const outputPermanentAddress = await page.locator('#output #permanentAddress').innerText();
```
Comment (Den4ik699) [inline tests/testCases/positiveTC.spec.js:20]: wrap this locators in Page Object

**Review by Den4ik699 — CHANGES_REQUESTED** [review]

File: tests/demoQA/pages/pageObject.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+        await textBoxIndicator('#userName').fill(name);
+        await textBoxIndicator('#userEmail').fill(email);
+        await textBoxIndicator('#currentAddress').fill(address1);
+        await textBoxIndicator('#permanentAddress').fill(address2);
+
+    }
+
+    async pressSubmit() {
+        await this.page.click('#submit');
```
Comment (Den4ik699) [inline tests/demoQA/pages/pageObject.js:23]: wrap button to Page Object

File: tests/testCases/negativeTC.spec.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
+const { test, expect } = require('@playwright/test');
+import { mainPage } from '../demoQA/pages/pageObject';
+import { filler } from '../demoQA/pages/inputs';
+
+test('Negative scenario', async({page}) => {
+   
+    const pageQA = new mainPage(page);
+
+    await pageQA.goto();
```
Comment (Den4ik699) [inline tests/testCases/negativeTC.spec.js:9]: lets move it to beforeEach hook and combine 2 tests in one file

File: tests/testCases/positiveTC.spec.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+import { mainPage } from '../demoQA/pages/pageObject';
+import { filler } from '../demoQA/pages/inputs';
+
+test ('Positive scenario', async ({page}) => {
+    
+    const pageQA = new mainPage(page);
+
+
+
+    await pageQA.goto();
```
Comment (Den4ik699) [inline tests/testCases/positiveTC.spec.js:11]: lets move it to beforeEach hook

**Review by Den4ik699 — CHANGES_REQUESTED** [review]

**Review by Den4ik699 — APPROVED** [review]
